lib/kafka_rest_client.js

```diff
@@ -100,16 +103,29 @@ KafkaRestClient.prototype.getProduceInterval = function getProduceInterval(retry
 KafkaRestClient.prototype.scheduleTopicMapRefresh = function scheduleTopicMapRefresh() {
     var self = this;
     if (self.proxyRefreshTime > 0) {
-        // console.log('Trying to schedule topic refresh job every: ' +
-        // self.proxyRefreshTime / 1000 + ' seconds.');
         /* eslint-disable no-undef,block-scoped-var */
         self.refreshIntervalId = setInterval(function refreshTopics() {
+            self.scheduleConnect(0);
+        }, self.proxyRefreshTime);
+        /* eslint-enable no-undef,block-scoped-var */
+    }
+};
+
+// Reconnect to rest proxy if scheduleTopicMapRefresh fails
+KafkaRestClient.prototype.scheduleConnect = function scheduleConnect(retry) {
+    var self = this;
+    if (retry < connectWaitTimeMs.length) {
```

Comment:
@proflayton For the issue you mentioned, I fixed that in: https://github.com/uber-common/node-kafka-rest-client/pull/32